Mathlib/Algebra/MonoidAlgebra/Basic.lean

```diff
@@ -281,30 +278,33 @@ section Semiring
 variable [Semiring k] [Monoid G]
 
 instance semiring : Semiring (MonoidAlgebra k G) :=
-  { MonoidAlgebra.nonUnitalSemiring,
-    MonoidAlgebra.nonAssocSemiring with
-    one := 1
-    mul := (· * ·)
-    zero := 0
-    add := (· + ·) }
+  { toNonUnitalSemiring := MonoidAlgebra.nonUnitalSemiring,
+    natCast := MonoidAlgebra.nonAssocSemiring.natCast
+    natCast_zero := MonoidAlgebra.nonAssocSemiring.natCast_zero
+    natCast_succ := MonoidAlgebra.nonAssocSemiring.natCast_succ
+    one_mul := MonoidAlgebra.nonAssocSemiring.one_mul
+    mul_one := MonoidAlgebra.nonAssocSemiring.mul_one }
 #align monoid_algebra.semiring MonoidAlgebra.semiring
 
 variable [Semiring R]
 
 /-- `liftNC` as a `RingHom`, for when `f x` and `g y` commute -/
 def liftNCRingHom (f : k →+* R) (g : G →* R) (h_comm : ∀ x y, Commute (f x) (g y)) :
     MonoidAlgebra k G →+* R :=
-  { liftNC (f : k →+ R) g with
-    toFun := liftNC (f : k →+ R) g
-    map_one' := liftNC_one _ _
-    map_mul' := fun _a _b => liftNC_mul _ _ _ _ fun {_ _} _ => h_comm _ _ }
+  -- { toMonoidHom :=liftNC (f : k →+ R) g with
+  { toMonoidHom := {
+      toFun := liftNC (f : k →+ R) g
+      map_one' := liftNC_one _ _
+      map_mul' := fun _a _b => liftNC_mul _ _ _ _ fun {_ _} _ => h_comm _ _ }
```

Comment:
Doesn't this generate an identical term?

Reply:
I don't think so but didn't check